pkg/experiment/query_backend/client/client.go

```diff
@@ -15,8 +15,8 @@ type Client struct {
 	grpcClient queryv1.QueryBackendServiceClient
 }
 
-func New(address string, grpcClientConfig grpcclient.Config) (*Client, error) {
-	conn, err := dial(address, grpcClientConfig)
+func New(address string, grpcClientConfig grpcclient.Config, dialOpts ...grpc.DialOption) (*Client, error) {
```

Comment:
Seems like the right thing to do to expose dial options.

e.g., `pkg/experiment/ingestor/client` does the same thing.

Reply:
Yes, I believe this is the right thing to do.

Also, you may want to init `grpcclient.Config` with the default values (in the same way it's done in `ingester/client`):

```go
grpcClientConfig.RegisterFlags(flag.NewFlagSet("", flag.PanicOnError))
```

(this, however, could be a separate test case)